pdata/internal/generated_immutable_slice.go

```diff
@@ -0,0 +1,132 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by "model/internal/cmd/pdatagen/main.go". DO NOT EDIT.
+// To regenerate this file run "go run model/internal/cmd/pdatagen/main.go".
+
+package internal
+
+// ImmutableByteSlice represents a []byte slice that cannot be mutated.
+type ImmutableByteSlice struct {
+	value []byte
+}
+
+// NewImmutableByteSlice creates a new ImmutableByteSlice by moving an underlying array of the provided []byte slice.
+// Note that the argument value will be cleared after the call,
+// use NewImmutableByteSliceFromCopy if it needs to be preserved.
+func NewImmutableByteSlice(val *[]byte) ImmutableByteSlice {
```

Comment:
My slight concern is if this may be a bit dangerous if the caller is not careful and tries to work with the nil-ed pointer after the call. However I don't think this is going to be called with a pointer variable, it will be most likely called by taking an address of any existing slice, so it should be safe, there is no way to cause a nil pointer access and panic.
So, I think this is fine. A bit non-idiomatic for Go, but I think it should be ok.